### PR TITLE
Fix sha256 for v3.4.0

### PR DESCRIPTION
**Pull request by mrakitin**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mrakitin** [comment]
@conda-forge-admin, please rerender

**Comment by mrakitin** [comment]
@conda-forge/core, could you please assist us with a weird issue observed in the CI builds, which I cannot reproduce while building with the same recipe locally/via Docker? Thanks in advance!

```py
ERROR :: get_rpaths_raw()=[] and patchelf=['$ORIGIN'] disagree for /home/conda/feedstock_root/build_artifacts/pyepics_1564417655156/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/epics/clibs/linux64/libca.so :: 
ERROR :: get_rpaths_raw()=[] and patchelf=['$ORIGIN'] disagree for /home/conda/feedstock_root/build_artifacts/pyepics_1564417655156/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/epics/clibs/linux32/libComPYEPICS.so :: 
ERROR :: get_rpaths_raw()=[] and patchelf=['$ORIGIN'] disagree for /home/conda/feedstock_root/build_artifacts/pyepics_1564417655156/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/epics/clibs/linuxarm/libComPYEPICS.so :: 
ERROR :: get_rpaths_raw()=[] and patchelf=['$ORIGIN'] disagree for /home/conda/feedstock_root/build_artifacts/pyepics_1564417655156/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/epics/clibs/linux32/libca.so :: 
ERROR :: get_rpaths_raw()=[] and patchelf=['$ORIGIN'] disagree for /home/conda/feedstock_root/build_artifacts/pyepics_1564417655156/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/epics/clibs/linuxarm/libca.so :: 
ERROR :: get_rpaths_raw()=[] and patchelf=['$ORIGIN'] disagree for /home/conda/feedstock_root/build_artifacts/pyepics_1564417655156/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/epics/clibs/linux64/libComPYEPICS.so :: 
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 445, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 436, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2343, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1642, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1002, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 867, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1150, in post_build
    check_overlinking(m, files)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1087, in check_overlinking
    m.config.enable_static)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 999, in check_overlinking_impl
    build_prefix, build_prefix_substitution)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 686, in _collect_needed_dsos
    envroot=run_prefix)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/liefldd.py", line 948, in __call__
    value = self.func(*args, **kw)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/liefldd.py", line 977, in get_linkages_memoized
    recurse=recurse, sysroot=sysroot, envroot=envroot, arch=arch)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/liefldd.py", line 481, in get_linkages
    sysroot=sysroot, arch=arch)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/pyldd.py", line 1145, in inspect_linkages
    filename, sysroot=sysroot, arch=arch)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/pyldd.py", line 1089, in _inspect_linkages_this
    cf = codefile(ReadCheckWrapper(f), arch)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/pyldd.py", line 1006, in codefile
    return elffile(file, list(initial_rpaths_transitive))
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/pyldd.py", line 877, in __init__
    es.postprocess(self, file)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/os_utils/pyldd.py", line 803, in postprocess
    end = n + strsec.table[n:].index('\0')
ValueError: substring not found
```

**Comment by mrakitin** [comment]
OK, there were 2 problems:
- missing `pip` in the run dependencies (that resolved Windows and OSX builds)
- an attempt in the `setup.py` to manage `.dylib` files for Darwin, which was resolved by the patch from @tacaswell, which we use for our channel at NSLS-II.

They are both resolved now. Reviews/comments are welcome. Thanks!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,21 +10,20 @@ source:
...
+     - libca.patch
 
 build:
-  number: 0
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  number: 1
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
 
 requirements:
   build:
```
Comment (mariusvniekerk) [inline recipe/meta.yaml]: Can this be?

```suggestion
  host:
```

**Comment by mariusvniekerk** [comment]
Is it even remotely possible to build the .so files on conda-forge?  Feels kinda weird vendoring them like this

**Comment by mrakitin** [comment]
@mariusvniekerk, thank you for having a look. Unfortunately, we don't have an ability to influence how pyepics developers want to distribute their libraries/binaries. @tacaswell has more thoughts about that.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,21 +10,20 @@ source:
...
 
 build:
-  number: 0
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  number: 1
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
 
 requirements:
-  build:
+   host:
```
Comment (mrakitin) [inline recipe/meta.yaml]:
```suggestion
  host:
```

**Comment by mariusvniekerk** [comment]
Merging.  Opened #13 to account for the issue

**Comment by mrakitin** [comment]
Thank you, @mariusvniekerk!

**Pull request merged**